contracts/auction.sol

```diff
@@ -158,3 +158,41 @@ contract AssertiveAuction is Assertive, UsingAuctionDatabase {
     }
 }
 
+contract AuctionFrontend is EventfulAuction
+                          , AssertiveAuction
+                          , SplittingAuction
+                          , FallbackFailer
+                          , AuctionFrontendType
+{
+    // Place a new bid on a specific auctionlet.
+    function bid(uint auctionlet_id, uint bid_how_much) {
+        assertBiddable(auctionlet_id, bid_how_much);
+        doBid(auctionlet_id, msg.sender, bid_how_much);
+        Bid(auctionlet_id);
+    }
+    // Allow parties to an auction to claim their take.
+    // If the auction has expired, individual auctionlet high bidders
+    // can claim their winnings.
+    function claim(uint auctionlet_id) {
+        assertClaimable(auctionlet_id);
+        doClaim(auctionlet_id);
```

Comment:
Events are minimal so far. We can expand as needed - I was leaving this to frontend input.

I didn't put `Claim` in as it doesn't cause an auction state change.

`claim` is only possible when the auctionlet has 'expired', after which point only the highest bidder can claim anyway.